packages/amplify-category-geo/src/service-walkthroughs/placeIndexWalkthrough.ts

```diff
@@ -0,0 +1,196 @@
+import _ from 'lodash';
+import uuid from 'uuid';
+import inquirer from 'inquirer';
+import { merge } from '../service-utils/resourceUtils';
+import { DataSourceIntendedUse, PlaceIndexParameters } from '../service-utils/placeIndexParams';
+import { apiDocs, ServiceName } from '../service-utils/constants';
+import { $TSContext } from 'amplify-cli-core';
+import { getCurrentPlaceIndexParameters } from '../service-utils/placeIndexUtils';
+import { getGeoServiceMeta, updateDefaultResource } from '../service-utils/resourceUtils';
+import { resourceAccessWalkthrough, pricingPlanWalkthrough, dataProviderWalkthrough } from './resourceWalkthrough';
+import { DataProvider, PricingPlan } from '../service-utils/resourceParams';
+
+/**
+ * Starting point for CLI walkthrough that creates a place index resource
+ * @param context The Amplify Context object
+ * @param parameters The configurations of the place index Resource
+ */
+export const createPlaceIndexWalkthrough = async (
+  context: $TSContext,
+  parameters: Partial<PlaceIndexParameters>
+): Promise<Partial<PlaceIndexParameters>> => {
+  // get the place index name
+  parameters = merge(parameters, await placeIndexNameWalkthrough(context));
+
+  // get the access
+  parameters = merge(parameters, await resourceAccessWalkthrough(parameters, ServiceName.PlaceIndex));
+
+  // optional advanced walkthrough
+  parameters = await placeIndexAdvancedWalkthrough(context, parameters);
+
+  // ask if the place index should be set as a default. Default to true if it's the only place index
+  const currentPlaceIndexResources = await getGeoServiceMeta(ServiceName.PlaceIndex);
+  if (currentPlaceIndexResources && Object.keys(currentPlaceIndexResources).length > 0) {
+    parameters.isDefault = await context.amplify.confirmPrompt('Do you want to set this search index as default?', true)
+  }
+  else {
+      parameters.isDefault = true;
+  }
+
+  return parameters;
+};
+
+export const placeIndexNameWalkthrough = async (context: any): Promise<Partial<PlaceIndexParameters>> => {
+    const indexNamePrompt = {
+        type: 'input',
+        name: 'name',
+        message: 'Provide a name for the location search index (place index):',
+        validate: context.amplify.inputValidation({
+            operator: 'regex',
+            value: '^[a-zA-Z0-9]+$',
+            onErrorMsg: 'You can use the following characters: a-z A-Z 0-9',
+            required: true,
+        }),
+        default: () => {
+            const [shortId] = uuid().split('-');
+            return `placeindex${shortId}`;
+        },
+    };
+    return await inquirer.prompt([indexNamePrompt]);
```

Comment:
I would suggest
```suggestion
    return {name: (await inquirer.prompt([indexNamePrompt])).name};
```
to make the structure of the returned object more obvious